BadUploads and BadDownloads tests are failing

Unit tests fail with the following output:

    --- FAIL: TestSuite/TestBadUploads (0.01s)
    	fixture.go:135: 
    		Test:     (*F).TestBadUploads()
    		(0):      /workspace/src/flywheel.io/sdk/tests/upload_test.go:32
    		Expected: '{"status_code": 404, "message": "The resource could not be found."}'
    		Actual:   '{"status_code": 404, "message": "The resource could not be found.", "request_id": "6d3a5691-1511891657"}'
    		(Should be equal)
    		
    --- FAIL: TestSuite/TestBadDownloads (0.00s)
    	fixture.go:135: 
    		Test:     (*F).TestBadDownloads()
    		(0):      /workspace/src/flywheel.io/sdk/tests/download_test.go:32
    		Expected: '{"status_code": 404, "message": "The resource could not be found."}'
    		Actual:   '{"status_code": 404, "message": "The resource could not be found.", "request_id": "91e87a79-1511891657"}'
    		(Should be equal)

This is due to direct string comparison of the error result string - The `request_id` field was added in scitran/core#1002.

We should be more robust in our error comparison.

--- a/tests/download_test.go
+++ b/tests/download_test.go
@@ -29,7 +29,7 @@ func (t *F) TestBadDownloads() {
 
 	// Could improve this in the future
 	err := <-result
-	t.So(err.Error(), ShouldEqual, "{\"status_code\": 404, \"message\": \"The resource could not be found.\"}")
+	t.So(err.Error(), ShouldMatchRegex, "\\{\"status_code\": 404, \"message\": \"The resource could not be found.\"\\, \"request_id\": \"[^\"]+\"}")
 	t.So(buffer.String(), ShouldEqual, "")
 }
 

--- a/tests/upload_test.go
+++ b/tests/upload_test.go
@@ -28,8 +28,10 @@ func (t *F) TestBadUploads() {
 	poem := "Are full of passionate intensity."
 	source = UploadSourceFromString("yeats.txt", poem)
 	_, result = t.UploadSimple("not-an-endpoint", nil, source)
+
 	// Could improve this in the future
-	t.So((<-result).Error(), ShouldEqual, "{\"status_code\": 404, \"message\": \"The resource could not be found.\"}")
+	err := <-result
+	t.So(err.Error(), ShouldMatchRegex, "\\{\"status_code\": 404, \"message\": \"The resource could not be found.\"\\, \"request_id\": \"[^\"]+\"}")
 }
 
 // Given an upload function, container ID, filename, and content - upload & check length

--- a/tests/z-util.go
+++ b/tests/z-util.go
@@ -7,6 +7,7 @@ import (
 	"io"
 	"io/ioutil"
 	"math/rand"
+	"regexp"
 	"strings"
 	"time"
 
@@ -80,6 +81,35 @@ func ShouldBeSameTimeAs(actual interface{}, expected ...interface{}) string {
 		return fmt.Sprintf(shouldBeTimeEqual, actualTime, expectedTime, actualTime.Sub(expectedTime))
 	}
 
+	return success
+}
+
+// Helper function to compare a string against a regular expression
+const (
+	invalidRegex     = "Invalid match expression '%s': %s"
+	regexBadMatch    = "Expected '%s' to match pattern: '%s'"
+	shouldUseStrings = "You must provide string instances as arguments to this assertion."
+)
+
+func ShouldMatchRegex(actual interface{}, pattern ...interface{}) string {
+	if fail := need(1, pattern); fail != success {
+		return fail
+	}
+
+	actualString, firstOk := actual.(string)
+	patternString, secondOk := pattern[0].(string)
+
+	if !firstOk || !secondOk {
+		return shouldUseStrings
+	}
+
+	matched, err := regexp.MatchString(patternString, actualString)
+	if err != nil {
+		return fmt.Sprintf(invalidRegex, patternString, err.Error())
+	}
+	if !matched {
+		return fmt.Sprintf(regexBadMatch, actualString, patternString)
+	}
 	return success
 }
 

